lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -297,7 +265,35 @@ void LocalEnforcer::aggregate_records(
           record.rule_id(), record.bytes_tx(), record.bytes_rx(), uc);
     }
   }
-  notify_finish_report_for_sessions(session_map, session_update);
+  complete_termination_for_released_sessions(
+      session_map, sessions_with_active_flows, session_update);
+}
+
+void LocalEnforcer::complete_termination_for_released_sessions(
```

Comment:
So when has to be terminated is marked as `SESSION_RELEASED`. Then, on the next pipelined report iteration, we will go over all the sessions looking for those that are `SESSION_RELEASED` and do not have any active flow to terminate.

What happen with the ones that have been makred as `SESSION_RELEASED` but have active flows?

Reply:
Yes, so the rule report from PipelineD what enforcement-stat flows still exist in PipelineD. So if session is RELEASED and we did NOT receive a rule usage, then that means the enforcement-stat flows attached to that Session has been cleared. 
If session is marked as released AND we received a rule update, we will leave the session as RELEASED. We can check again on the next rule update.